linter/linter.go

```diff
@@ -581,6 +605,32 @@ func (l *Linter) lintSubRoutineDeclaration(decl *ast.SubroutineDeclaration, ctx
 	return types.NeverType
 }
 
+func (l *Linter) lintPenaltyboxDeclaration(decl *ast.PenaltyboxDeclaration) types.Type {
+	// validate penaltybox name
+	if !isValidName(decl.Name.Value) {
+		l.Error(InvalidName(decl.Name.GetMeta(), decl.Name.Value, "penaltybox").Match(PENALTYBOX_SYNTAX))
+	}
+
+	if len(decl.Block.Statements) > 0 {
+		l.Error(NonEmptyPenaltyboxBlock(decl.GetMeta(), decl.Name.Value))
+	}
+
+	return types.NeverType
+}
+
+func (l *Linter) lintRatecounterDeclaration(decl *ast.RatecounterDeclaration) types.Type {
+	// validate ratecounter name
+	if !isValidName(decl.Name.Value) {
+		l.Error(InvalidName(decl.Name.GetMeta(), decl.Name.Value, "ratecounter").Match(RATECOUNTER_SYNTAX))
+	}
+
+	if len(decl.Block.Statements) > 0 {
+		l.Error(NonEmptyRatecounterBlock(decl.GetMeta(), decl.Name.Value))
```

Comment:
Please link the rule matching that you added to `RATECOUNTER_SYNTAX `.
(Or, an Individual rule like `RATECOUNTER_MUST_BE_EMPTY` is fine, maybe)


```suggestion
		l.Error(NonEmptyRatecounterBlock(decl.GetMeta(), decl.Name.Value).Match(RATECOUNTER_SYNTAX))
```